radicle-tracker/src/thread.rs

```diff
@@ -0,0 +1,827 @@
+use nonempty::NonEmpty;
+use thiserror::Error;
+
+/// The "liveness" status of some data.
+///
+/// TODO: we may want to consider `Modified`.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum Status<A> {
+    /// The data has been created.
+    Live(A),
+    /// The data has been created and also deleted.
+    Dead(A),
+}
```

Comment:
This type is a bit confusing, especially given the comment: It seems to represent the data *and* the status, not just a status, even though the type is called `Status`. :thinking: I guess I would expect a type like:
```
struct Data<A> {
  status: Status,
  data: A,
}
```
It would also avoid having to match on the status to get the data. :man_shrugging:

Reply:
Well, those things are equivalent, but I see your point. I'll think about it :thinking:

Reply:
Yeah if you want to keep the type as-is I'd probably sill suggest a better name and to fix the comment :)